### ISSUE TITLE
Event Upload POST out of bounds exception

### ISSUE THREAD

**Issue by xiphirx** [amplitude/Amplitude-Android#80]
We're seeing a lot of this crash using v2.5.0. Not sure if its with this SDK or OkHttp. We use OkHttp 2.7.4, which is the latest 2.x release.

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=2048; index=2048
       at okio.Buffer.readByte(Buffer.java:280)
       at com.squareup.okhttp.HttpUrl.canonicalize(HttpUrl.java:1594)
       at com.squareup.okhttp.FormEncodingBuilder.add(FormEncodingBuilder.java:38)
       at com.amplitude.api.AmplitudeClient.makeEventUploadPostRequest(AmplitudeClient.java:886)
       at com.amplitude.api.AmplitudeClient$8.run(AmplitudeClient.java:816)
       at android.os.Handler.handleCallback(Handler.java:808)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:193)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by benjamin-bader** [comment]
This is likely not an Amplitude problem.  Okio has a bit of a threading issue, where the global SegmentPool (an internal data structure) can become corrupted if a Buffer is written to by multiple threads anywhere in the process.  Once that happens, arbitrary buffer operations can fail.

**Comment by djih** [comment]
@benjamin-bader do you know if this is fixed in OkHttp 3? We have an unreleased sdk version with OkHttp 3, maybe that would help

**Comment by benjamin-bader** [comment]
It's an open issue in okio (which OkHttp3 depends on), so I very much doubt it.

**Comment by djih** [comment]
@benjamin-bader ah ok thanks for the info. Here are the Okio issues in question: https://github.com/square/okio/issues/134 and https://github.com/square/okio/issues/190

**Comment by djih** [comment]
@benjamin-bader did they ever fix it? The 2 Okio issues I linked have been closed. Do you know if there is anything we need to do on our end (like pull a new version or something)

**Comment by benjamin-bader** [comment]
Nope, it's deferred at least until okio 1.8.  The issue tracking the actual fix is square/okio#163.
